main.cpp

```diff
@@ -39,7 +31,8 @@ int main(int argc, char *argv[]) {
 
     std::cout << std::endl;
 
-    delete h1, h2;
+    delete h1;
+    delete h2;
```

Comment:
Itt megvan a felszabadítás, de más kilépési pontja is van a programnak, ahol maradhat fel nem szabadított memória. Nyilván ha úgyis kilép a program, akkor nem akkora probléma, de azért ha már ez az aktuális feladat témája, akkor ne lehessen sehol memóriaszivárgás.